src/app/resources/resources-add.component.ts

```diff
@@ -0,0 +1,196 @@
+  'Literature', 'Math', 'Music', 'Politics and Government', 'Reference',
+  'Religion', 'Science', 'Social Sciences', 'Sports', 'Technology' ];
+  levels = new FormControl();
+  levelList = [ 'Early Education', 'Lower Primary', 'Upper Primary',
+  'Lower Secondary', 'Upper Secondary', 'Undergraduate', 'Graduate', 'Professional' ];
+  mediums = [ 'Text', 'Graphic/Pictures', 'Audio/Music/Book', 'Video' ];
+  openWith = [ 'Just download', 'HTML', 'PDF.js', 'Bell-Reader', 'MP3', 'Flow Video Player', 'BeLL Video Book Player', 'Native Video' ];
+  resourceType = [ 'Textbook', 'Lesson Plan', 'Activities', 'Exercises', 'Discussion Questions' ];
+  todaydate = new Date();
+  file: any;
+  mediaType: any;
+  resourceForm: FormGroup;
+  readonly dbName = 'resources'; // make database name a constant
+
+  constructor(
+    private location: Location,
+    private router: Router,
+    private fb: FormBuilder,
+    private couchService: CouchService,
+    private resourceValidatorService: ResourceValidatorService,
+    private userService: UserService
+  ) {
+    this.createForm();
+  }
+
+  createForm() {
+    this.resourceForm = this.fb.group({
+      title: [
+        '',
+        Validators.required,
+        // an arrow function is for lexically binding 'this' otherwise 'this' would be undefined
+        ac => this.resourceValidatorService.checkResourceExists$(ac)
+      ],
+      author: '',
+      year: '',
+      language: '',
+      publisher: '',
+      linkToLicense: '',
+      subject: [ '', Validators.required ],
+      level: [ '', Validators.required ],
+      openWith: '',
+      resourceFor: [],
+      medium: '',
+      articleDate: '',
+      resourceType: '',
+      addedBy: '',
+      openUrl: [],
+      openWhichFile: ''
+    });
+  }
+
+  onSubmit() {
+    if (this.resourceForm.valid) {
+      if (this.file !== undefined) {
+        if (this.file.type.indexOf('image') > -1) {
+          this.mediaType = 'image';
+        } else if (this.file.type.indexOf('pdf') > -1) {
+          this.mediaType = 'pdf';
+        } else if (this.file.type.indexOf('audio') > -1) {
+          this.mediaType = 'audio';
+        } else if (this.file.type.indexOf('video') > -1) {
+          this.mediaType = 'video';
+        } else if (this.file.type.indexOf('zip') > -1) {
+          this.zipAttachment();
+        } else {
+          this.mediaType = 'other';
+        }
+        if (this.file.type.indexOf('zip') === -1 ) {
+          const reader = new FileReader(),
+          rComp = this;
+          reader.readAsDataURL(this.file);
+          reader.onload = () => {
+          // FileReader result has file type at start of string, need to remove for CouchDB
+          const fileData = reader.result.split(',')[1],
+          attachments = {};
+          attachments[rComp.file.name] = {
+            content_type: rComp.file.type,
+            data: fileData
+          };
+          const resource = Object.assign({ },
```

Comment:
You do not need to use `Object.assign` when assigning an object.  `Object.assign` can be used to shallow copy objects and/or combine two objects together.  When creating an object you can just assign that object (PR to follow).